### PR TITLE
fix: prevent withdrawal of zero amount to avoid unnecessary fees

### PR DESCRIPTION
**Pull request by abel533**
Added check in onSubmit to prevent withdrawal if amount is zero

---

For more details, open the [Copilot Workspace session](https://copilot-workspace.githubnext.com/RSSNext/Follow?shareId=789051f6-2228-4c4f-ac48-a593d371605d).

### CODE REVIEW

File: apps/renderer/src/modules/power/my-wallet-section/withdraw.tsx [hunk 1]
```diff
@@ -69,6 +69,9 @@ const WithdrawModalContent = ({ dismiss }: { dismiss: () => void }) => {
   })
 
   const onSubmit = (values: z.infer<typeof formSchema>) => {
+    if (values.amount === 0) {
```
Comment (hyoban) [inline apps/renderer/src/modules/power/my-wallet-section/withdraw.tsx:72]: We should perform data validation in form schema

**Review by hyoban — CHANGES_REQUESTED** [review]

File: apps/renderer/src/modules/power/my-wallet-section/withdraw.tsx (hunk 1 above)
Comment (abel533) [reply apps/renderer/src/modules/power/my-wallet-section/withdraw.tsx:72]:
```js
  const formSchema = z.object({
    address: z.string().startsWith("0x").length(42),
    amount: z.number().min(0).max(cashablePowerTokenNumber),
  })
```
Modify min value directly? How much is appropriate?

Comment (hyoban) [reply apps/renderer/src/modules/power/my-wallet-section/withdraw.tsx:72]: try `z.number().positive()`

**Comment by hyoban** [comment]
Thanks for your contribution!

**Pull request merged**